NOTICE

```diff
@@ -2,4 +2,15 @@ Apache Flink Agents
 Copyright 2025 The Apache Software Foundation
 
 This product includes software developed at
-The Apache Software Foundation (http://www.apache.org/).
\ No newline at end of file
+The Apache Software Foundation (http://www.apache.org/).
+
+Apache Flink
+Copyright 2014-2025 The Apache Software Foundation
+
+This project contains the following code copied from Apache Flink Stateful Functions:
+runtime/src/main/java/org/apache/flink/agents/runtime/common/*
+runtime/src/main/java/org/apache/flink/agents/runtime/feedback/*
+runtime/src/main/java/org/apache/flink/agents/runtime/logger/*
+runtime/src/main/java/org/apache/flink/agents/runtime/memory/*
+runtime/src/main/java/org/apache/flink/agents/runtime/queue/*
+Original Copyright: 2014-2020 The Apache Software Foundation
```

Comment:
1. Let's remove the detailed directories. We need to keep the NOTICE file as concise as possible. And we are likely to make further changes to the copied codes, so the listed 'xx/*' directories will also contain non-copied codes.
2. I think the copyright years of statefun is 2014-2023, as suggested by its [NOTIE file](https://github.com/apache/flink-statefun/blob/master/NOTICE). This is also aligned with its commit history (last commit in Sep 2023).

```suggestion
Apache Flink Stateful Functions
Copyright: 2014-2023 The Apache Software Foundation
```